From 1507ed73dbf5dfc330f7954f81a5ae47bbc3b4e7 Mon Sep 17 00:00:00 2001
From: b07117 <b07117@freescale.com>
Date: Wed, 11 May 2011 23:07:29 +0800
Subject: [PATCH 142/635] ENGR00143243, 800x600 second display show horizontal
 flicker when UI change

GPU request 4K align physical address for second display resizing

Signed-off-by: xianzhong <b07117@freescale.com>
---
 libgralloc/framebuffer.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/libgralloc/framebuffer.cpp b/libgralloc/framebuffer.cpp
index a889215..37d0a41 100755
--- a/libgralloc/framebuffer.cpp
+++ b/libgralloc/framebuffer.cpp
@@ -904,8 +904,8 @@ static int mapSecFrameBuffer(fb_context_t* ctx)
     info.blue.length    = 5;
     info.transp.offset  = 0;
     info.transp.length  = 0;
-    info.yres_virtual = info.yres * NUM_BUFFERS;
-	info.xres_virtual = info.xres;
+    info.yres_virtual = ALIGN_PIXEL_128(info.yres) * NUM_BUFFERS;
+    info.xres_virtual = ALIGN_PIXEL(info.xres);
                         
     if (ioctl(sec_fp, FBIOPUT_VSCREENINFO, &info) == -1) {
         LOGE("Error!Second display FBIOPUT_VSCREENINFO");
@@ -1177,7 +1177,7 @@ static int resizeToSecFrameBuffer_c2d(int base,int phys,fb_context_t* ctx)
                 
     if((ctx->mRotate == 0)||(ctx->mRotate == 180))
     {
-        if(ctx->sec_disp_w/ctx->sec_disp_h >= ctx->device.width/ctx->device.height){
+        if(ctx->sec_disp_w >= ctx->sec_disp_h*ctx->device.width/ctx->device.height){
             dstRect.width = ctx->sec_disp_h*ctx->device.width/ctx->device.height;
         }
         else{
@@ -1185,7 +1185,7 @@ static int resizeToSecFrameBuffer_c2d(int base,int phys,fb_context_t* ctx)
         }
     }
     else{
-        if(ctx->sec_disp_w/ctx->sec_disp_h >= ctx->device.height/ctx->device.width){
+        if(ctx->sec_disp_w >= ctx->sec_disp_h*ctx->device.height/ctx->device.width){
             dstRect.width = ctx->sec_disp_h*ctx->device.height/ctx->device.width;
         }
         else{
@@ -1285,7 +1285,7 @@ void * secDispShowFrames(void * arg)
             ctx->sec_info.yoffset = 0; 
         }
         else{
-            ctx->sec_info.yoffset = ctx->sec_disp_h;
+            ctx->sec_info.yoffset = ctx->sec_info.yres_virtual/NUM_BUFFERS;
         }
 
         ctx->sec_disp_next_buf = !ctx->sec_disp_next_buf;  
-- 
1.8.0

